pkgs/native_toolchain_c/lib/src/native_toolchain/apple_clang.dart

```diff
@@ -43,6 +49,15 @@ final Tool appleLd = Tool(
       wrappedResolver: appleClang.defaultResolver!,
       relativePath: Uri.file('ld'),
     ),
+    PathToolResolver(
+      toolName: 'Apple linker',
+      executableName: OS.current.executableFileName('ld'),
```

Comment:
Does this not have false positives?

Please add tests on Linux and Windows that appleLd doesn't resolve to any instances.